### PR TITLE
Throw CannotResolveType error when compiler cannot resolve it

### PR DESCRIPTION
**Pull request by yayaa**
**Description**:
If a return type is not resolved, motif will not be able to check whether it has `Scope` annotation or not meaning the following conditions are not correct either. As in, if it is a scope but motif was not able to detect that because it couldn't resolve the return type - it can still report a child scope as `AccessMethodParameters`. This change should prompt a more relevant error message.

### CODE REVIEW

File: errormessage/src/main/kotlin/motif/errormessage/CannotResolveTypeHandler.kt [hunk 1]
```diff
@@ -0,0 +1,17 @@
+package motif.errormessage
+
+import motif.models.CannotResolveType
+
+class CannotResolveTypeHandler(private val error: CannotResolveType) : ErrorHandler {
```
Comment (TonyTangAndroid) [inline errormessage/src/main/kotlin/motif/errormessage/CannotResolveTypeHandler.kt:5]: So far, I only know that we need to add license header like [this new added file](https://github.com/uber/motif/pull/227/files#diff-e95cdc9b4b6771bd0aa35517e4d54da58d25a9321014068b82e4ada9157209fc) to make its corresponding unit test pass.

Comment (yayaa) [reply errormessage/src/main/kotlin/motif/errormessage/CannotResolveTypeHandler.kt]: Thanks, it seems like running the tests locally, auto-generates and resolves that issue :)

**Comment by yayaa** [comment]
In the first implementation, I was throwing CannotResolveType exception if return type cannot be resolved but then the following tests were failing. Looking into them, found out that they had only one thing in common as they had access methods with *primitive return types*, hence **limited throwing exception only to non-primitive types**.

- T060_default_objects_methods2_JAVA
- T040_keywords_JAVA
- E027_invalid_scope_method_JAVA
- T075_scope_factory_unused_dependencies_JAVA
- T049_binds_primitive_JAVA
- T060_default_objects_methods2_KOTLIN
- T040_keywords_KOTLIN
- T075_scope_factory_unused_dependencies_KOTLIN
- T049_binds_primitive_KOTLIN

**Comment by tyvsmith** [comment]
Let's have a test added

**Comment by yayaa** [comment]
> Let's have a test added

@tyvsmith thought about it 🙈 but since it requires a type that cannot be resolved, I couldn't find how to create it. Probably requires several modules with visibilities maybe? Or any other idea how to test?

**Comment by tyvsmith** [comment]
> > Let's have a test added
> 
> @tyvsmith thought about it, but since it requires a type that cannot be resolved, I couldn't find how to create it. Probably requires several modules with visibilities maybe? Or any other idea how to test?
> 
> I tried just adding a package that does not exist, but that triggers compilation error
> 
> ```
> import notvisible_package.SomeScope;
> @motif.Scope
> public interface Scope {
>     SomeScope someScope();
> }
> ```
> 
> ```
> Scope.java:23: error: cannot find symbol
>     SomeScope someScope();
>     ^
>   symbol:   class SomeScope
>   location: interface Scope
> e: /.../motif/compiler/../tests/src/main/java/testcases/E060_cannot_resolve_type/Scope.java:21: error: [Processor:MiscError] motif.compiler.Processor was unable to process this interface because not all of its dependencies could be resolved. Check for compilation errors or a circular dependency with generated code.
> public interface Scope {
> ```

How about we try having a dependency on a provided time dep that we try to reference? Or could we do this with a type created via reflection?

**Comment by TonyTangAndroid** [comment]
@tyvsmith ,  just synced with @yayaa offline, while the test code is pending, is there a way to publish a snapshot version of this pull request and try to validate the effectiveness against our monorepo. I do have code that could stably reproduce the unexpected `[ACCESS METHOD PARAMETERS]` error.

**Comment by TonyTangAndroid** [comment]
I managed to test it out the error message and real use cases against our internal monorepo, and it works out as expected.  Please refer to the attached screenshot for reference.
Screen Shot 2022-02-02 at 3 06 41 AM

**Comment by TonyTangAndroid** [comment]
Synced with @yayaa offline, we are to refine the suggested message to be potentially easier to be  understood.

**Comment by TonyTangAndroid** [comment]
Here is a refined version of error message with @yayaa updated pull request. I would say that now it is in a solid state to move foward.

Screen Shot 2022-02-02 at 10 13 21 AM

**Review by TonyTangAndroid — APPROVED** [review]
Thank you for all the effort and patience working on this pull request.

**Comment by tyvsmith** [comment]
Talked to @TonyTangAndroid and we will explore testing this with runtimeOnly deps in the future.

**Pull request merged**